### ISSUE TITLE
Mermaid config doesn't work.

### ISSUE THREAD

**Issue by Erifirin** [slidevjs/slidev#1850]
I tried to change mermaid theme as it described in documentation: https://sli.dev/custom/config-mermaid. I created a file ./setup/mermaid.ts and copied example into it. But these changes did not affect anything and mermaid still uses the default theme. However, custom theme variables work properly if mermaid.ts is changed while slidev is running. After restarting slidev, mermaid reverts to default theme again.

**Minimal reproduction**

1. Create ./setup/mermaid.ts with content from example https://sli.dev/custom/config-mermaid#custom-theme-styles;
2. Run slidev;
3. Go to slide with mermaid graph/flowchart.

**Environment**

- Slidev: 0.49.27, 0.49.29
- Browser: chrome 128
- OS: win 10

**Comment by KermanX** [comment]
possibly #1830

**Comment by Erifirin** [comment]
No, I have one instance of slidev running from terminal. Moreover, I've faced the same behavior on macos.

os: macos 14.6.1 (M1, 2020)
browser: safari
node: 22.7.0
slidev: 0.49.29

Here's what I do:

1. `mkdir slides`
2. `cd slides`
3. `npm init slidev@latest`
    Project name: test
    Install and start it now?: yes
    Choose the package manager: npm
4. Wait for presentation to open. Go to slide 12.
5. Create ./setup/mermaid.ts with following (taken from mermaid examples)
```ts
import { defineMermaidSetup } from '@slidev/types'

export default defineMermaidSetup(() => {
  return {
    theme: 'base',
    themeVariables: {
      primaryColor: '#BB2528',
      primaryTextColor: '#fff',
      primaryBorderColor: '#7C0000',
      lineColor: '#F8B229',
      secondaryColor: '#006100',
      tertiaryColor: '#fff'
    }
  }
})
```
6. Go back to the slides. Nothing has changed.
![image](https://github.com/user-attachments/assets/f5ee7615-bafe-46c0-bab9-6bc1330705b9)

7. Ok. Return to terminal with slidev and press `r` to restart.
8. In browser go to slide 12 again.
9. Make any minor edits to the file ./setup/mermaid.ts and save.
10. Go to slide 12. Wow! All the arrows turned yellow, but rest remained the same:
![image](https://github.com/user-attachments/assets/c1b33450-6c71-4538-9ba1-705fd526bb24)

12. Now let's stop slidev by pressing `q`.
13. And start it again:
`cd test`
`npm exec slidev`
14. Go to slide 12 and it again uses default theme:
![image](https://github.com/user-attachments/assets/f277e1a5-bf62-4e47-8e84-684ab1a4abf0)


In result, the only working way to change the mermaid theme is to write themeVariables in each mermaid block. It works perfect but very inconvenient.

```
%%{
  init: {
    'theme': 'base',
    'themeVariables': {
      'primaryColor': '#BB2528',
      'primaryTextColor': '#fff',
      'primaryBorderColor': '#7C0000',
      'lineColor': '#F8B229',
      'secondaryColor': '#006100',
      'tertiaryColor': '#fff'
    }
  }
}%%
```

![image](https://github.com/user-attachments/assets/4a5eecc3-bd2e-4185-9608-677e4342f8b8)

**Comment by Erifirin** [comment]
@KermanX It seems I've managed to resolve it.
It's necessary to force empty a browser cache after creating file `./setup/mermaid.ts`. Now `defineMermaidSetup()` method is called every time the slidev is launched. As a result, the colors of the diagrams change, although not completely. In light theme everything works well, but in dark theme only part of themeVariables are applied :)

![image](https://github.com/user-attachments/assets/4324e839-5386-42b9-8bf4-05e0455d11fe)
![image](https://github.com/user-attachments/assets/f476458e-02ae-41f1-b641-73b7cb69b721)

**Comment by LeTuR** [comment]
Same issue as @Erifirin, I managed to make custom `setup/mermaid.ts`  theme configuration work by enforcing light mode with `colorSchema: light`.
My custom theme is a meant to be a dark theme. I need to override all default light theme to match default dark config.

**Comment by ajmoralesa** [comment]
Same issues here, despite of the `setup/mermaid.ts` file, the only change taking pace is theme but not the other variables

**Comment by yuttie** [comment]
For anyone with a similar setup—having multiple Markdown files in a directory and running something like `npx slidev slides/slides-01.md`—I was able to resolve the issue by placing the setup directory next to the Markdown files (i.e., at slides/setup).
